apps/assistants/models.py

```diff
@@ -76,6 +87,33 @@ def supports_file_search(self):
     def tools_enabled(self):
         return len(self.tools) > 0
 
+    @transaction.atomic()
+    def create_new_version(self, *args, **kwargs):
+        from .sync import push_assistant_to_openai
+
+        version_number = self.version_number
+        self.version_number = version_number + 1
+        self.save(update_fields=["version_number"])
+        assistant_version = super().create_new_version(save=False, *args, **kwargs)
+        assistant_version.version_number = version_number
+        assistant_version.save()
+
+        assistant_version.files.set(self.files.all())
```

Comment:
I think we'd want to duplicate the files rather? People won't be able to edit the file from the other assistant, but they would be able to delete it from there. There is a `duplicate` method on the File model which seems useful here

Reply:
https://github.com/dimagi/open-chat-studio/pull/833/commits/a21c1fd2094048a45d5e0cab589aba481e2eb226